File: code/community/1304604811022307332/realtime-vs-postprocess-transcription-accuracy.go
package main

import (
	"bytes"
	"encoding/json"
	"fmt"
	"net/http"
	"os"
)

func main() {
	// Configure your Deepgram API Key via an environment variable
	apiKey := os.Getenv("DEEPGRAM_API_KEY")

	// Set the Deepgram endpoint for pre-recorded transcription
	url := "https://api.deepgram.com/v1/listen"

	// Specify your audio source for transcription
	audioURL := "https://example.com/audio.wav"

	// Prepare the request body
	requestBody, _ := json.Marshal(map[string]string{
		"url": audioURL,
	})

	// Create a new HTTP request
	req, _ := http.NewRequest("POST", url, bytes.NewBuffer(requestBody))
	req.Header.Set("Authorization", fmt.Sprintf("Token %s", apiKey))
	req.Header.Set("Content-Type", "application/json")

	// Make the request
	client := &http.Client{}
	resp, err := client.Do(req)
	if err != nil {
		panic(err)
	}
	defer resp.Body.Close()

	// Read the response
	var result map[string]interface{}
	json.NewDecoder(resp.Body).Decode(&result)

	// Access and print the transcription result
	fmt.Println("Final transcription:", result) // Access the correct fields based on the actual JSON structure
}

File: code/community/1304604811022307332/realtime-vs-postprocess-transcription-accuracy.cs
using System;
using System.Net.Http;
using System.Threading.Tasks;

class Program
{
    private static readonly HttpClient client = new HttpClient();

    static async Task Main()
    {
        // Configure your Deepgram API Key via an environment variable
        var apiKey = Environment.GetEnvironmentVariable("DEEPGRAM_API_KEY");

        if (apiKey == null)
        {
            Console.WriteLine("Please set the DEEPGRAM_API_KEY environment variable");
            return;
        }

        // Set the Deepgram endpoint for pre-recorded transcription
        var url = "https://api.deepgram.com/v1/listen";

        // Specify your audio source for transcription
        var audioUrl = "https://example.com/audio.wav";

        var content = new StringContent($"{{ \"url\": \"{audioUrl}\" }}", System.Text.Encoding.UTF8, "application/json");
        client.DefaultRequestHeaders.Add("Authorization", $"Token {apiKey}");

        var response = await client.PostAsync(url, content);
        var responseString = await response.Content.ReadAsStringAsync();

        Console.WriteLine("Final transcription: " + responseString);
    }
}